### ISSUE TITLE
Http2Stream.OnReset assertion failed exception running test on macOS: OSX.1013.Amd64.Open.

### ISSUE THREAD

**Issue by danmosemsft** [dotnet/corefx#40115]
```
Process terminated. Assertion failed.
   at System.Net.Http.Http2Connection.Http2Stream.OnReset(Exception resetException, Nullable`1 resetStreamErrorCode, Boolean canRetry) in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs:line 658
   at System.Net.Http.Http2Connection.AbortStreams(Exception abortException) in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs:line 1439
   at System.Net.Http.Http2Connection.Abort(Exception abortException) in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs:line 1369
   at System.Net.Http.Http2Connection.FlushOutgoingBytesAsync() in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs:line 186
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine) in /_/src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/AsyncMethodBuilder.cs:line 1015
   at System.Net.Http.Http2Connection.FlushOutgoingBytesAsync()
   at System.Net.Http.Http2Connection.EndWrite(Boolean forceFlush) in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs:line 778
   at System.Net.Http.Http2Connection.FinishWrite(FlushTiming flush) in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs:line 751
   at System.Net.Http.Http2Connection.SendRstStreamAsync(Int32 streamId, Http2ProtocolErrorCode errorCode) in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs:line 867
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine) in /_/src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/AsyncMethodBuilder.cs:line 1015
   at System.Net.Http.Http2Connection.SendRstStreamAsync(Int32 streamId, Http2ProtocolErrorCode errorCode)
   at System.Net.Http.Http2Connection.Http2Stream.Reset() in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs:line 284
   at System.Net.Http.Http2Connection.Http2Stream.CancelResponseBody() in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs:line 345
   at System.Net.Http.Http2Connection.Http2Stream.Cancel() in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs:line 320
   at System.Net.Http.Http2Connection.Http2Stream.ReadResponseHeadersAsync(CancellationToken cancellationToken) in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs:line 779
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in /_/src/System.Private.CoreLib/shared/System/Threading/ExecutionContext.cs:line 172
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread) in /_/src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/AsyncMethodBuilder.cs:line 617
   at System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox box, Boolean allowInlining) in /_/src/System.Private.CoreLib/shared/System/Threading/Tasks/TaskContinuation.cs:line 794
   at System.Threading.Tasks.Task.RunContinuations(Object continuationObject) in /_/src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs:line 3319
   at System.Threading.Tasks.Task.FinishStageThree() in /_/src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs:line 2176
   at System.Threading.Tasks.Task.CancellationCleanupLogic() in /_/src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs:line 3151
   at System.Threading.Tasks.Task.TrySetCanceled(CancellationToken tokenToRecord, Object cancellationException) in /_/src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs:line 3266
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception exception) in /_/src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/AsyncMethodBuilder.cs:line 784
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.SetException(Exception exception) in /_/src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/AsyncMethodBuilder.cs:line 283
   at System.Net.Http.Http2Connection.Http2Stream.GetCancelableWaiterTask(CancellationToken cancellationToken) in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs:line 1044
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state) in /_/src/System.Private.CoreLib/shared/System/Threading/ExecutionContext.cs:line 289
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread) in /_/src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/AsyncMethodBuilder.cs:line 625
   at System.Threading.ThreadPoolWorkQueue.Dispatch() in /_/src/System.Private.CoreLib/shared/System/Threading/ThreadPool.cs:line 663
./RunTests.sh: line 161: 25287 Abort trap: 6           "$RUNTIME_PATH/dotnet" exec --runtimeconfig System.Net.Http.Functional.Tests.runtimeconfig.json xunit.console.dll System.Net.Http.Functional.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=nonnetcoreapptests -notrait category=nonosxtests -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing $RSP_FILE
```
Since thisi s macOS, we do not have a dump (I think @mjanecke is working on that). Also no good clue which test failed.

https://helix.dot.net/api/2019-06-17/jobs/0fef5713-a52e-40f7-8542-641b0bdb491c/workitems/System.Net.Http.Functional.Tests/console

**Comment by stephentoub** [comment]
https://github.com/dotnet/corefx/blob/3cdd1f49a13e61060e24c8221ec381ce4cc3fc94/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs#L658
cc: @geoffkizer

**Comment by karelz** [comment]
Fixed in 3.0 in PR #40180 and in master (5.0) in PR #40123
